js_modules/dagster-ui/packages/ui-core/src/overview/OverviewRoot.tsx

```diff
@@ -33,16 +31,7 @@ export const OverviewRoot = () => {
         path="/overview/sensors"
         render={() => (flagLegacyNav ? <OverviewSensorsRoot /> : <Redirect to="/automation" />)}
       />
-      <Route
-        path="/overview/automation"
-        render={() =>
-          !flagLegacyNav && automaterializeSensorsFlagState !== 'has-global-amp' ? (
```

Comment:
Here's the source of the bug. When the query for the flag state isn't complete, the flag state value is `unknown`, and we redirect. By moving this check into the route itself, we can wait until the flag state is ready before deciding whether to redirect.